### DISCUSSION TITLE
Entity ID different in PickingEvent::Clicked than what I originally stored when using glTF

### DISCUSSION THREAD

**Discussion by Newbytee** [aevyrie/bevy_mod_picking#197] [Q&A] [answered]
I'm making a small game with Bevy as an exercise to learn this engine, and as I need to detect when a user clicks on an object in the world with their mouse, I went looking for how to do that in Bevy and found this plugin! I'm using glTF, so I quickly ran into what's described in https://github.com/aevyrie/bevy_mod_picking/discussions/163 and used the solution at https://github.com/aevyrie/bevy_mod_picking/discussions/163#discussioncomment-4717813 to resolve it. However, after doing this, I noticed that the entities inside the `PickingEvent::Clicked` event had different IDs (I hope that's the right terminology) than the ones I had stored away in an array on creation. See https://github.com/Newbytee/bevy_picking_mre/blob/70fe5d49c57d370eab9495ae217e09129eef6fbd/src/main.rs#L110-L125

Do you have any ideas regarding what I might be doing wrong here, and how I can fix this? What I really need is some way to associate the `Piece` it's stored in so I can update some state there (which is not present in this stripped-down example). Maybe I'm thinking about this the wrong way?

Link to full "minimal" reproducible example here: https://github.com/Newbytee/bevy_picking_mre

**Comment by aevyrie** [comment, marked as answer]
Thanks for providing an MRE, sorry I haven't caught up to this sooner. I haven't fully debugged this, but it looks like the issue is that you are storing the `Entity` of the piece, and the picking plugin is hitting the mesh, which is a *child* of the root piece entity.

Assuming you are still interested in a solution, I would suggest taking a look at the event listener features that have been added recently. Specifically, bubbling will make this much easier to solve. In short, picking events automatically bubble up the hierarchy, allowing you to put event listeners with callbacks on the root `Piece`, and react when *any* of its children receives a picking event.
